src/libsyntax/feature_gate.rs

```diff
@@ -1891,6 +1895,27 @@ impl<'a> PostExpansionVisitor<'a> {
             Err(mut err) => err.emit(),
         }
     }
+
+    /// Recurse into all places where a `let` expression would be feature gated
+    /// and emit gate post errors for those.
+    fn find_and_gate_lets(&mut self, e: &'a ast::Expr) {
+        match &e.node {
+            ast::ExprKind::Paren(e) => {
+                self.find_and_gate_lets(e);
+            }
+            ast::ExprKind::Binary(op, lhs, rhs) if op.node == ast::BinOpKind::And => {
+                self.find_and_gate_lets(lhs);
+                self.find_and_gate_lets(rhs);
+            }
+            ast::ExprKind::Let(..) => {
+                gate_feature_post!(
+                    &self, let_chains, e.span,
+                    "`let` expressions in this position are experimental"
+                );
```

Comment:
Similarly to https://github.com/rust-lang/rust/pull/60669#discussion_r283101494, this feature is extending the grammar, so technically it needs to be gated at parse time.

Reply:
It certainly needs to be gated at parse time if `let` is an expression.

Reply:
Yes, I will fix that 👍

Reply:
Done.